dosed/utils/data_from_h5.py

```diff
@@ -5,22 +5,29 @@
 import h5py
 
 from ..preprocessing import normalizers
+from ..preprocessing.resamplers import resample
 
 
-def get_h5_data(filename, signals, downsampling_rate):
+def get_h5_data(filename, signals, resampling_fs):
     import time
     time.sleep(5)
+
+    signals_fs = [signal['fs'] for signal in signals]
+    downsampling_rates = [signal['fs'] / resampling_fs for signal in signals]
+
     with h5py.File(filename, "r") as h5:
-        # Check that all signals have the same size and sampling frequency
-        signals_size = set([int(h5[signal["h5_path"]].size) for signal in signals])
-        assert len(signals_size) == 1, "Different signal sizes found!"
-        signal_size = len(range(0, signals_size.pop(), downsampling_rate))
+
+        # Check that all signals have the same size after resampling  to target frequency
+        signals_size = set([int(h5[signal["h5_path"]].size / downsampling)
+                            for signal, downsampling in zip(signals, downsampling_rates)])
+        #assert len(signals_size) == 1, "Different signal sizes found ! {}".format(signals_size)
+        signal_size = len(range(0, min(signals_size)))
```

Comment:
Here you can create 

t_target = np.cumsum([1/fs] * min(signals_size))

And then below you just create a one liner

t_source = np.cumsum([1/fs] * min(signal["fs"]))
data[i, :] == interp1d(t_source, h5[signal["h5_path"]][:])(t_target)